src/coreclr/jit/lsra.cpp

```diff
@@ -9575,14 +9561,24 @@ void LinearScan::TupleStyleDump(LsraTupleDumpMode mode)
                         printf("  Dummy def of V%02u at #%d\n", interval->varNum, currentRefPosition->rpNum);
                         break;
                     case RefTypeBB:
+                        if (printedBlockHeader)
+                        {
+                            continueLoop = false;
+                            break;
+                        }
                         block->dspBlockHeader(compiler);
                         printedBlockHeader = true;
                         printf("=====\n");
                         break;
                     default:
-                        printf("Unexpected RefPosition type at #%d\n", currentRefPosition->rpNum);
+                        continueLoop = false;
                         break;
                 }
+                if (!continueLoop)
+                {
+                    // Avoid loop iteration that will update currentRefPosition
```

Comment:
Why not just keep the `break` that is at the beginning of the loop?

Reply:
> `jit-analyze` output doesn't show up. @jakobbotsch do you know why?

```scala

[10:03:16] Total bytes of delta: -36 (-0.00% of base)

[10:03:16] Generated asm is located under C:\h\w\A34A08D0\p\artifacts\spmi\asm.coreclr_tests.run.Linux.arm64.checked\base C:\h\w\A34A08D0\p\artifacts\spmi\asm.coreclr_tests.run.Linux.arm64.checked\diff

[10:03:16] No textual differences. Is this an issue with coredistools?

[10:03:16] 88 contexts with differences found (1 improvements, 0 regressions)

```

Hmm, it is a combination of #53773 and my recent change in #76238 exacerbating this problem. I think that during full replay of all contexts, we are more likely to see the problem #53773, while when we create individual disassembly files we are less likely to see it (due to less churn on the memory allocator and so on). So since we are basing the diff numbers on the first run only now we see the problem more often.

@EgorBo do you think there is a chance we can merge #76112 with the main functionality disabled? Or merge the part of it that fixes #53773?

Reply:
The complicated expressions were one of the motivations for the PR (the `&` iterator thing was the other).  When I looked at simplifying them, I noticed that a few included the same set of cases as the logic in the rest of the loop (in this case, the list of RefTypeExpUse, RefTypeDummyDef, and RefTypeBB appears twice), so I tried to combine them.

The `printf` in the default case was another clue - it seems to be more like an assertion that we shouldn't get there because the original check is the same as the switch.

I'm open to suggestions for other ideas.  That I have some asm diffs both encourages me to do the cleanup since I must have misread the code somewhere but also worries me a bit about churn.  Is there an easy way to verify that #73424 didn't have any asm diffs?  (The job seems to be gone)

Reply:
That makes sense and cleaner. I was just curious about the reasoning.